dio/lib/src/response.dart

```diff
@@ -32,6 +32,16 @@ class Response<T> {
   /// Http status code.
   int? statusCode;
 
+  /// Returns if the response has an error
+  ///
+  /// `hasError` is true when status code is not between 200 and 299
+  bool get hasError => statusCode != null && statusCode! ~/ 100 != 2;
```

Comment:
IIRC we have `ValidateStatus` as a prediction, it might be inappropriate to use such a getter.

Reply:
I thought it would be nice to have the getter direct to the `Response`object, but if it is an appropriate way I'll remove the getter. Thanks :)